### DISCUSSION TITLE
Get the client ip from a `Request`

### DISCUSSION THREAD

**Discussion by ssendev** [tokio-rs/axum#968] [Q&A] [answered]
I'm currently migrating a rocket app to axum and had previously used [client_ip](https://github.com/SergioBenitez/Rocket/blob/7bbe0457a5ecc9d4b3ca0a2bc6120ffbe1e7febd/core/lib/src/request/request.rs#L413) but can't find a way to get the ip with axum

**Comment by jplatte** [comment, marked as answer]
Have a look at [`Router::into_make_service_with_connect_info`](https://docs.rs/axum/latest/axum/struct.Router.html#method.into_make_service_with_connect_info).

**Reply by wjcroom** [reply]
is a server  ip

**Comment by wjcroom** [comment]
https://github.com/pol-inc/axum-client-ip/tree/main  this may do
